test/Scrutor.Tests/ScanningTests.cs

```diff
@@ -140,7 +175,7 @@ public void AutoRegisterAsMatchingInterface()
                     .AsMatchingInterface()
                     .WithTransientLifetime());
 
-            Assert.Equal(2, Collection.Count);
+            Assert.Equal(3, Collection.Count);

```

Comment:
I see what you mean. Does register as self makes sense when you have an interface on it. Isn't the goal of those services to use interfaces/inheritance. for the more complicated use cases where you have more than one interface specified I see anyway no other way than adding `ServiceDescriptor` for all the cases. But my change would break the current behavior and user would have to set the `type` to the current type :(. 

Hmm.... Complicated stuff....